libraries/iio/iio.c

```diff
@@ -1225,10 +1225,7 @@ ssize_t iio_init(struct iio_desc **desc, struct iio_init_param *init_param)
 
 	ldesc->phy_type = init_param->phy_type;
 	if (init_param->phy_type == USE_UART) {
-		ret = uart_init((struct uart_desc **)&ldesc->uart_desc,
-				init_param->uart_init_param);
-		if (IS_ERR_VALUE(ret))
-			goto free_desc;
+		ldesc->uart_desc = init_param->uart_desc;
```

Comment:
Good catch, didnt see it, will remove it !
As for calling uart_remove in app, that's not a problem for these reasons:
1. the project commits have changes in them so that the projects compile - it is not catastrophic not to remove uart because the project already failed for some other reason if it got to the "error" label.
2. all those projects will eventually use iio_app which will handle the situation properly